Use waylandsink in Wayland sessions for GStreamer 1.16+

Integral-Tech opened the pull request:
Use waylandsink instead of xvimagesink/ximagesink in Wayland sessions.

fduncanh committed:
Add uxplay rendering of audio cover art: uxplay -ca (w/o filename)

fduncanh committed:
reset video when coverart rendering terminates

fduncanh commented:
Thanks,  we will  look into this.  Does autovideosink fail to choose waylandsink when appropriate?

fduncanh commented:
Has waylandsink been improving with time?   Would it be appropriate to only make waylandsink the default choice if gstreamer version is sufficiently new?

Integral-Tech committed:
Use waylandsink in Wayland sessions for GStreamer 1.16+

Use waylandsink instead of xvimagesink/ximagesink in Wayland sessions
for GStreamer 1.16+.

Integral-Tech commented:
@fduncanh Now `waylandsink` is selected by default for GStreamer 1.16+  :)

fduncanh commented:
@Integral-Tech 
> Now waylandsink is selected by default for GStreamer 1.16+ :)

does this mean the fix is not actually needed?


If needed, this code below should be done in renderers/video_renderer.c  (video_renderer_init)  and renderers/x_display_fix.h

```
   guint major, minor, micro, nano;


    gst_version (&major, &minor, &micro, &nano);


    if (major >= 1 && minor >= 16) {


    	const char *xdg_session_type = getenv("XDG_SESSION_TYPE");


    	if (xdg_session_type && strcmp(xdg_session_type, "wayland") == 0) {


            videosink = "waylandsink";


    	}


    }

```

fduncanh commented:
This is really a gstreamer autovideosink issue, though... maybe there is a gstreamer fix?
